### PR TITLE
feat: [SRM-13456]: refactoring Metric Thresholds and tests for metric thresholds

### PR DESCRIPTION
**Pull request by harshilbits**
### Summary
1.Refactoring Metric Thresholds and tests for metric thresholds

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by harshilbits** [comment]
retrigger titlecheck

**Comment by harshilbits** [comment]
retrigger titlecheck

**Review by praneshg239 — DISMISSED** [review]

**Review by deepesh-ui — DISMISSED** [review]

**Comment by harshilbits** [comment]
retrigger coverage

**Comment by harshilbits** [comment]
retrigger jest

**Comment by harshilbits** [comment]
retrigger jest

**Review by praneshg239 — APPROVED** [review]

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/MetricThresholds/MetricThresholdContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+export interface MetricThresholdContainerProps {
+  healthSourceConfig: HealthSourceConfig
+  groupedCreatedMetrics: GroupedCreatedMetrics
+  isMetricThresholdEnabled: boolean
+}
+
+export default function MetricThresholdContainer(props: MetricThresholdContainerProps): JSX.Element {
+  const { healthSourceConfig, groupedCreatedMetrics, isMetricThresholdEnabled } = props
+  const { values: formValues } = useFormikContext<CommonHealthSourceConfigurations>()
+  const isShowMetricThreshold = getCanShowMetricThresholds({
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/MetricThresholds/MetricThresholdContainer.tsx:20]: shall we wrap in useMemo ?

**Review by deepesh-ui — APPROVED** [review]

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/MetricThresholds/MetricThresholdContainer.tsx (hunk 1 above)
Comment (harshilbits) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/MetricThresholds/MetricThresholdContainer.tsx:20]: Can be done. Can be taken in coming PRs

**Pull request merged**